pkg/executables/cmk.go

```diff
@@ -119,39 +139,156 @@ func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domain string,
+func (c *Cmk) ValidateZonesPresent(ctx context.Context, zones []v1alpha1.CloudStackZoneRef) ([]v1alpha1.CloudStackResourceIdentifier, error) {
+	var zoneIdentifiers []v1alpha1.CloudStackResourceIdentifier
+	for _, z := range zones {
+		zone := z.Zone
+		var filterString string
+		if zone.Type == v1alpha1.Id {
+			filterString = fmt.Sprintf("id=\"%s\"", zone.Value)
+		} else {
+			filterString = fmt.Sprintf("name=\"%s\"", zone.Value)
+		}
+		commandArgs := []string{"list", "zones"}
+		filterArgs := []string{filterString}
+		commandArgs = append(commandArgs, filterArgs...)
+		result, err := c.exec(ctx, commandArgs...)
+		if err != nil {
+			return nil, fmt.Errorf("error getting zones info: %v", err)
+		}
+		if result.Len() == 0 {
+			return nil, fmt.Errorf("zone %s not found", filterString)
+		}
+
+		response := struct {
+			CmkZones []cmkZone `json:"zone"`
+		}{}
+		if err = json.Unmarshal(result.Bytes(), &response); err != nil {
+			return nil, fmt.Errorf("failed to parse response into json: %v", err)
+		}
+		cmkZones := response.CmkZones
+		if len(cmkZones) > 1 {
+			return nil, fmt.Errorf("duplicate zone %s found", filterString)
+		} else if len(zones) == 0 {
+			return nil, fmt.Errorf("zone %s not found", filterString)
+		} else {
+			zoneIdentifiers = append(zoneIdentifiers, v1alpha1.CloudStackResourceIdentifier{Name: cmkZones[0].Name, Id: cmkZones[0].Id})
+		}
+	}
+	return zoneIdentifiers, nil
+}
+
+func (c *Cmk) ValidateDomainPresent(ctx context.Context, domain string) (v1alpha1.CloudStackResourceIdentifier, error) {
+	domainIdentifier := v1alpha1.CloudStackResourceIdentifier{Name: domain, Id: ""}
+	result, err := c.exec(ctx, "list", "domains", fmt.Sprintf("name=\"%s\"", domain))
+	if err != nil {
+		return domainIdentifier, fmt.Errorf("error getting domain info: %v", err)
+	}
+	if result.Len() == 0 {
+		return domainIdentifier, fmt.Errorf("domain %s not found", domain)
+	}
+
+	response := struct {
+		CmkDomains []cmkDomain `json:"domain"`
+	}{}
+	if err = json.Unmarshal(result.Bytes(), &response); err != nil {
+		return domainIdentifier, fmt.Errorf("failed to parse response into json: %v", err)
+	}
+	domains := response.CmkDomains
+	if len(domains) > 1 {
+		return domainIdentifier, fmt.Errorf("duplicate domain %s found", domain)
+	} else if len(domains) == 0 {
+		return domainIdentifier, fmt.Errorf("domain %s not found", domain)
+	}
+
+	domainIdentifier.Id = domains[0].Id
+	domainIdentifier.Name = domains[0].Name
+
+	return domainIdentifier, nil
+}
+
+func (c *Cmk) ValidateNetworkPresent(ctx context.Context, domainId string, zoneRef v1alpha1.CloudStackZoneRef, zones []v1alpha1.CloudStackResourceIdentifier, account string, multipleZone bool) error {
+	commandArgs := []string{"list", "networks"}
+	var filterArgs []string
+	if zoneRef.Network.Type == v1alpha1.Id {
+		filterArgs = append(filterArgs, fmt.Sprintf("id=\"%s\"", zoneRef.Network.Value))
+	}
+	if multipleZone {
+		filterArgs = append(filterArgs, fmt.Sprintf("type=\"%s\"", "Shared"))
```

Comment:
Concatenating string literal with string literal?

Reply:
In-fact, "Shared" should be a constant. It has semantics in the context of cloud monkey so deserves dedicated constant.